packages/edit-site/src/hooks/commands/use-common-commands.js

```diff
@@ -0,0 +1,167 @@
+				},
+			},
+		];
+	}, [ canReset, onReset ] );
+
+	return {
+		isLoading: false,
+		commands,
+	};
+}
+
+function useGlobalStylesOpenCssCommands() {
+	const { openGeneralSidebar, setEditorCanvasContainerView } = unlock(
+		useDispatch( editSiteStore )
+	);
+	const history = useHistory();
+	const { canEditCSS } = useSelect( ( select ) => {
+		const { getEntityRecord, __experimentalGetCurrentGlobalStylesId } =
+			select( coreStore );
+
+		const globalStylesId = __experimentalGetCurrentGlobalStylesId();
+		const globalStyles = globalStylesId
+			? getEntityRecord( 'root', 'globalStyles', globalStylesId )
+			: undefined;
+
+		return {
+			canEditCSS:
+				!! globalStyles?._links?.[ 'wp:action-edit-css' ] ?? false,
+		};
+	}, [] );
+
+	const commands = useMemo( () => {
+		if ( ! canEditCSS ) {
+			return [];
+		}
+
+		return [
+			{
+				name: 'core/edit-site/open-styles-css',
+				label: __( 'Open CSS' ),
+				icon: styles,
+				callback: ( { close } ) => {
+					close();
+					history.push( {
+						path: '/wp_global_styles',
+						canvas: 'edit',
+					} );
+					openGeneralSidebar( 'edit-site/global-styles' );
+					setEditorCanvasContainerView( 'global-styles-css' );
+				},
+			},
+		];
+	}, [
+		history,
+		openGeneralSidebar,
+		setEditorCanvasContainerView,
+		canEditCSS,
+	] );
+	return {
+		isLoading: false,
+		commands,
+	};
+}
+
+export function useCommonCommands() {
+	const { openGeneralSidebar, setEditorCanvasContainerView } = unlock(
+		useDispatch( editSiteStore )
+	);
+	const { set } = useDispatch( preferencesStore );
+	const history = useHistory();
+
+	useCommand( {
+		name: 'core/edit-site/open-global-styles-revisions',
+		label: __( 'Open styles revisions' ),
+		icon: backup,
+		callback: ( { close } ) => {
+			close();
+			history.push( {
+				path: '/wp_global_styles',
+				canvas: 'edit',
+			} );
+			openGeneralSidebar( 'edit-site/global-styles' );
+			setEditorCanvasContainerView( 'global-styles-revisions' );
+		},
+	} );
+
+	useCommand( {
+		name: 'core/edit-site/open-styles',
+		label: __( 'Open styles' ),
+		callback: ( { close } ) => {
+			close();
+			history.push( {
+				path: '/wp_global_styles',
+				canvas: 'edit',
+			} );
+			openGeneralSidebar( 'edit-site/global-styles' );
+		},
+		icon: styles,
+	} );
+
+	useCommand( {
+		name: 'core/edit-site/toggle-styles-welcome-guide',
+		label: __( 'Learn about styles' ),
+		callback: ( { close } ) => {
+			close();
+			history.push( {
+				path: '/wp_global_styles',
+				canvas: 'edit',
+			} );
+			openGeneralSidebar( 'edit-site/global-styles' );
+			set( 'core/edit-site', 'welcomeGuideStyles', true );
+			// sometimes there's a focus loss that happens after some time
+			// that closes the modal, we need to force reopening it.
+			setTimeout( () => {
+				set( 'core/edit-site', 'welcomeGuideStyles', true );
+			}, 500 );
+		},
+		icon: help,
+	} );
+
+	useCommandLoader( {
+		name: 'core/edit-site/reset-global-styles',
+		hook: useGlobalStylesResetCommands,
+	} );
```

Comment:
Why does _Reset styles to default_ use a command loader? Couldn't `useCommonCommands` call `useGlobalStylesReset` directly and then call `useCommand`? `useGlobalStylesReset` doesn't appear to do any asynchronous data loading.

Reply:
It's because of the `canReset` check. Sometimes the command need to be "hidden". I did notice this pattern already and it made me wonder whether we should replace it with an `isDisabled` option in the `useCommand` hook to unregister the command once disabled or something like that.